frontend/public/extend/kubevirt/components/vm-list.jsx

```diff
@@ -0,0 +1,138 @@
+import * as classNames from 'classnames';
+
+import { ListHeader, ColHead, List, ListPage, ResourceRow } from '../../../components/factory';
+import { Firehose, ResourceLink, ResourceKebab, Kebab } from '../../../components/utils';
+import { VirtualMachineModel, VirtualMachineInstanceModel, PodModel, NamespaceModel } from '../../../models';
+import { referenceForModel } from '../../../module/k8s';
+
+import { startStopVmModal } from './modals';
+import { getResourceKind, getLabelMatcher, findVMI, findPod, getFlattenForKind, getVmStatus } from '../module/k8s/vms';
+import * as s from '../strings';
+
+const VMHeader = (props) => <ListHeader>
+  <ColHead {...props} className="col-lg-2 col-md-2 col-sm-2 col-xs-6" sortField="metadata.name">Name</ColHead>
+  <ColHead {...props} className="col-lg-2 col-md-2 col-sm-2 col-xs-6" sortField="metadata.namespace">Namespace</ColHead>
+  <ColHead {...props} className="col-lg-2 col-md-2 col-sm-2 hidden-xs" sortField="spec.running">State</ColHead>
+  <ColHead {...props} className="col-lg-2 col-md-2 col-sm-2 hidden-xs" sortField="metadata.phase">Phase</ColHead>
+  <ColHead {...props} className="col-lg-2 col-md-2 col-sm-2 hidden-xs">Virtual Machine Instance</ColHead>
+  <ColHead {...props} className="col-lg-2 col-md-2 col-sm-2 hidden-xs">Pod</ColHead>
+</ListHeader>;
+
+const getAction = (vm) => {
+  return _.get(vm, 'spec.running', false) ? 'Stop Virtual Machine' : 'Start Virtual Machine';
+};
+
+const menuActionStart = (kind, vm) => ({
+  label: getAction(vm),
+  callback: () => startStopVmModal({
+    kind,
+    resource: vm,
+    start: !_.get(vm, 'spec.running', false),
+  }),
+});
+
+export const menuActions = [menuActionStart, Kebab.factory.Edit, Kebab.factory.Delete];
+
+export const StateColumn = (props) => {
+  let value;
+  if (props.loaded) {
+    const vm = props.flatten(props.resources);
+    if (vm) {
+      value = getVmStatus(vm);
+    }
+  }
+  return <span className={classNames({'text-muted': !value})}>{value || s.notAvailable}</span>;
+};
+
+export const PhaseColumn = (props) => {
+  let value;
+  if (props.loaded) {
+    const resources = props.flatten(props.resources);
+    const vmi = props.filter(resources);
+    value = _.get(vmi, 'status.phase');
+  }
+  return <span className={classNames({'text-muted': !value})}>{value || s.notAvailable}</span>;
+};
+
+export const FirehoseResourceLink = (props) => {
+  if (props.loaded) {
+    const data = props.flatten(props.resources);
+    if (data) {
+      let resource = data[0];
+      if (props.filter) {
+        resource = props.filter(data);
+      }
+      if (resource) {
+        const name = resource.metadata.name;
+        const namespace = resource.metadata.namespace;
+        const title = resource.metadata.uid;
+        const kind = resource.kind || PodModel.kind;
+        // TODO: if `kind` refers to CRD object, use `referenceForModel` function
+        return <ResourceLink kind={kind} name={name} namespace={namespace} title={title} />;
```

Comment:
I'd leave off title here. UID is not helpful to show on hover. (I know we do it other places).

Reply:
This (and a lot of other code) was taken from existing code, I'll remove the `title` as suggested.